.github/workflows/publish.yml

```
@@ -0,0 +1,44 @@
+name: Publish
+
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: read # to fetch code (actions/checkout)
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    name: Release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+
+      - name: Set node version to 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'pnpm'
+
+      - name: Prepare
+        run: pnpm install --frozen-lockfile
+
+      - name: Set publishing config
+        run: pnpm config set '//registry.npmjs.org/:_authToken' "${NPM_AUTH_TOKEN}"
+        env:
+          NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
+
+      - name: Publish
+        run: |
+          PACKAGE_DIST_TAG=$(node -e "console.log(/^\d+\.\d+\.\d+(\-(\w+)\.\d+)$/.exec(require('./package.json').version)?.[2] || 'latest')")
+          pnpm publish --access public --tag $PACKAGE_DIST_TAG
+
+      - name: Set next dist-tag
+        run: |
+          PACKAGE_VERSION=$(node -e "console.log(require('./package.json').version)")
+          pnpm dist-tag add @faker-js/faker@$PACKAGE_VERSION next
```

IMO we should tag it with next by default and set the latest only when it is a stable release:

```suggestion
          PACKAGE_VERSION=$(node -e "console.log(require('./package.json').version)")
          pnpm publish --access public --tag next
		  if [[ "$PACKAGE_VERSION" != *"-"* ]]; then
            pnpm dist-tag add @faker-js/faker@$PACKAGE_VERSION latest
          fi
```

Not tested.

---

No, we are extracting the dist-tag out of the version field and then e.g. alpha, beta or whatever is used
This is default tagging behavior guided to semver

---

confused: But we don't have alpha or beta tags!? Or did we remove them after release?

---

Oh good that we found out that we misused that before 🫤 
- https://www.npmjs.com/package/vite?activeTab=versions
- https://www.npmjs.com/package/typescript?activeTab=versions
- https://www.npmjs.com/package/react?activeTab=versions

just some examples that use e.g. beta dist-tags (which are not my own packages)